live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/storage/StorageClient.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.storage;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.URL;
+
+import org.apache.http.StatusLine;
+
+public interface StorageClient
```

Comment:
I think we should move this interface to domain and probably adjust it a little as it shows internal implementation details (use of apache and pre-signed urls) but it is probably something we can do later (I have no idea how we want to use it so idk how we can improve it😂)

Reply:
So this the issue at this stage - I'm not sure how it will look finally. It will clarify in the second part of this task. In general it's a representation of an service/component, so I'm not sure if `domain` is the right place. Overall I would leave it as it is, till the second part is done.